### ISSUE TITLE
GTM+GA Working Example / Documentation Request

### ISSUE THREAD

**Issue by rpavez** [alinemorelli/react-gtm#58]
Please create an example and write documentation on how setup GTM + GA
After lot of try/error was able to:

Here's a quick example might be useful for other people

1. Create a VirtualPageView
```        TagManager.dataLayer({
            dataLayer: {
                event: 'PageView',
                page: page,
            },
            dataLayerName: 'PageDataLayer',
        })
```
2. Submit an Event
```
            dataLayer: {
                event: 'GAEvent',
                category: 'Coverage',
                value: 'Test'
            },
            dataLayerName: 'PageDataLayer',
        })
```
The example to use events in the README is useless.

Also would be good to explain how get the data with GTM
1. Create a Tag for GA PageView and assigned Custom Event Trigger with name matching the datalayer eventname (i.e. PageView)
2. For GA Events create Variables of type "Data Layer Variable" with the key names of the data from the data layer, then create a Tag for GA and assign a custom event (i.e. GAEvent) and in the Tag config setup the fields by selecting from the created Variables.

**Comment by ooridoori** [comment]
Hey @rpavez thanks so much for the helpful notes. I'm actually still a bit confused on how to implement everything. 
Does the Tag Manager need to be "initialized" on every component that needs to push an event? If its possible do you think you can show an example of how you used these snippets in your code? 

I also am confused on the how to get the data part - do you think u can provide screenshots?

Many thanks 🙏

**Comment by rpavez** [comment]
Hi @ooridoori I can't provide screenshots but initialization is only needed once, I'm initializing on the index.js and then on each page I'm just using `TagManager.dataLayer` as shown on my example, I also implemented an HOC called withTracker like shown here https://github.com/react-ga/react-ga/wiki/React-Router-v4-withTracker so each page is wrapped with withTracker to submit a virtualPageView event when moving through the pages.

**Comment by dancedric** [comment]
Yea there aren't a lot of instructions in this module

**Comment by mwq27** [comment]
Thanks a lot for this example. I was just trying to figure out how to work with tracking events. With the ReactGA module we did:
```js
ReactGA.event({
        category: 'sales',
        action,
        label,
      });
```
But now it looks like that should be replaced with:
```js
TagManager.dataLayer({
            dataLayer: {
                event: 'GAEvent',
                category: 'sales',
                action,
                label,
            }
        })
```
Does that sound right?

**Comment by itayganor** [comment]
@mwq27 Is the event string has to be `GAEvent`?

**Comment by rpavez** [comment]
Not at all, you can choose whatever name you want that matched the config on TagManager

**Comment by gcosgreave-us** [comment]
Can the keys. in the object be anything you want? I am guessing as long as you have a GTM trigger set up to take that value?

**Comment by peculiarity** [comment]
It somehow doens't work for me

This is the initialization code

```javascript
 useEffect(() => {
    const tagManagerArgs = {
      gtmId: 'GTM-0000001', //masked
      dataLayerName: 'PageViewLayer'
    }

    TagManager.initialize(tagManagerArgs)   
  }, [])
```
And on each page load I call 
```javascript
TagManager.dataLayer({
      dataLayer: {
        event: 'PageView',
        page
      },
      dataLayerName: 'PageViewLayer',
 })
```

I have my tag a `Custom Event Trigger` with event name - `PageView` 
No google-analytics calls on my website
